xval/utils.q

```diff
@@ -0,0 +1,342 @@
+// xval/utils.q - Cross validation utilities
+// Copyright (c) 2021 Kx Systems Inc
+//
+// Utilities for cross validation library
+
+\d .ml
+
+// Cross validation indexing
+
+// @private
+// @kind function
+// @category xvUtility
+// @desc Shuffle data point indices
+// @param data {any} Table, matrix or list
+// @return {long[]} Indices of data shuffled
+xv.i.shuffle:{[data]
+  0N?count data
+  }
+
+// @private
+// @kind function
+// @category xvUtility
+// @desc Find indices required to split data into k-folds
+// @param k {int} Number of folds
+// @param data {any} Table, matrix or list
+// @return {long[][]} Indices required to split data into k sub-sets
+xv.i.splitIdx:{[k;data]
+  (k;0N)#til count data
+  }
+
+// @private
+// @kind function
+// @category xvUtility
+// @desc Find shuffled indices required to split data into k-folds
+// @param k {int} Number of folds
+// @param data {any} Table, matrix or list
+// @return {long[][]} Shuffled indices required to split data into k 
+//   sub-sets
+xv.i.shuffIdx:{[k;data]
+  (k;0N)#xv.i.shuffle data
+  }
+
+// @private
+// @kind function 
+// @category xvUtility
+// @desc Split target data ensuring that each distinct value appears in
```

Comment:
This should indicate not that we're saying that each distinct value will be represented but rather that the proportion of each distinct values in the test dataset will be represented in the train test split